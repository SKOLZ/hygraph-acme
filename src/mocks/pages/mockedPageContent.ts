interface Props {
  sectionAmount: number;
  blockAmount: number;
}

export const mockedPageContent = ({ sectionAmount, blockAmount }: Props) => ({
  sections: new Array(sectionAmount).fill("").map((_, sectionIndex) => ({
    id: sectionIndex,
    blocks: new Array(blockAmount).fill("").map((_, blockIndex) => ({
      __typename: "Hero",
      id: blockIndex,
      title: `Section ${sectionIndex} - Block ${blockIndex}`,
      description: {
        raw: {
          children: [
            {
              type: "paragraph",
              children: [
                {
                  text: "Equip your team with the industryâ€™s leading native GraphQL Content APIs, Content Federation capabilities and rich set of developer tools, to build truly composable applications.",
                },
              ],
            },
          ],
        },
      },
      buttonActions: [
        {
          id: "clt7bqmoz9dkc07ujrs01ghn3",
          buttonText: "Get started",
          size: "lg",
        },
      ],
    })),
  })),
});
