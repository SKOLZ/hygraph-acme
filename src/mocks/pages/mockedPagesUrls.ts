export const mockedPagesUrls = [
  {
    url: "/home",
  },
  {
    url: "/hello",
  },
];
