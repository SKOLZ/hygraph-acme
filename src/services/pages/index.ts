import { client } from "../client";
import { pageContentQuery } from "./queries/pageContentQuery";

export const getPageContent = () => client.request(pageContentQuery);
