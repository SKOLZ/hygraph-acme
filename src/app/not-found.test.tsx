import { render } from "@testing-library/react";
import NotFound from "./not-found";
import { mockedPagesUrls } from "@/mocks/pages/mockedPagesUrls";

jest.mock("../services/pages/", () => ({
  getPagesUrls: () => Promise.resolve(mockedPagesUrls),
}));

describe("not-found", () => {
  it("renders the not found page correctly", async () => {
    const jsx = await NotFound();
    const {container} = render(jsx);
    expect(container).toMatchSnapshot();
  })
});
