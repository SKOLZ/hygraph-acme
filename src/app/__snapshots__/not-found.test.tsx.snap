// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not-found renders the not found page correctly 1`] = `
<div>
  <div
    class="mx-auto flex max-w-screen-lg flex-1 flex-col items-center justify-center gap-4 text-center"
  >
    <h1
      class="title-1 text-secondary-900"
    >
      The page you were looking for doesn't exist
    </h1>
    <p
      class="text-2 text-secondary-400"
    >
      Maybe you were looking for one of these:
    </p>
    <ul
      class="flex gap-1"
    >
      <li>
        <a
          class="text-2 rounded-sm bg-secondary-100 px-2 py-1 text-secondary-600"
          href="/home"
        >
          /home
        </a>
      </li>
      <li>
        <a
          class="text-2 rounded-sm bg-secondary-100 px-2 py-1 text-secondary-600"
          href="/hello"
        >
          /hello
        </a>
      </li>
    </ul>
  </div>
</div>
`;
